- Description: zfcp: Reset host_scribble field in scsi_cmnd.
  Symptom:     Addressing exception, system hangs
  Problem:     zfcp relies on the fact that the host_scribble field for each scsi
               command that is passed to it is NULL. When the zfcp lldd cannot
               send a command it passes the command back to the scsi mid layer
               but does not reset this field to NULL. Result is that if this
               command gets retried an addressing exception may occur since the
               zfcp lldd assumes that host_scribble points to a valid address if
               it is not NULL.
  Solution:    Reset host_scribble field to NULL.
  Archs:       s390-31, s390-64

--- linux-2.5/drivers/s390/scsi/zfcp_fsf.c	30 Mar 2004 13:18:08 -0000	1.43
+++ linux-2.5/drivers/s390/scsi/zfcp_fsf.c	19 Apr 2004 09:26:32 -0000	1.43.2.1
@@ -29,7 +29,7 @@
  */
 
 /* this drivers version (do not edit !!! generated and updated by cvs) */
-#define ZFCP_FSF_C_REVISION "$Revision: 1.43 $"
+#define ZFCP_FSF_C_REVISION "$Revision: 1.43.2.1 $"
 
 #include "zfcp_ext.h"
 
@@ -3523,6 +3523,7 @@
 
 	zfcp_fsf_req_free(fsf_req);
 	fsf_req = NULL;
+	scsi_cmnd->host_scribble = NULL;
  success:
  failed_req_create:
 	write_unlock_irqrestore(&adapter->request_queue.queue_lock, lock_flags);

